src/metabase/query_processor/middleware/parameters/dates.clj

```diff
@@ -109,12 +109,15 @@
                :end   (t/minus dt (t/days 1))})
     :filter (fn [_ field] ["=" field ["relative_datetime" -1 "day"]])}
 
-   {:parser (regex->parser #"past([0-9]+)(day|week|month|year)s", [:int-value :unit])
+   ;; adding a tilde (~) at the end of a past<n><unit> filter means we should include the current day/etc.
```

Comment:
Can we have a test the makes sure this properly catches tildes?

Reply:
We can ;) see https://github.com/metabase/metabase/pull/6530/files#diff-c775b10cbaf781a636a6e6090f2309a8R79

Reply:
weird, I misread that test and never noticed the tilde